### DISCUSSION TITLE
Ray with Sail?

### DISCUSSION THREAD

**Discussion by imClement** [spatie/ray#101] [General]
Hello,

How is the use of Ray trough a Sail based environment?

Should we follow the instructions from "Configuring Ray -> When developing using Docker"?

Thank you,
Clement

**Comment by freekmurze** [comment]
I've not used Sail myself, I and I guess you should indeed allow the docker instructions.

**Comment by Tuginho** [comment]
Yes. just publish the config file and then set host to 'host.docker.internal'

**Comment by imClement** [comment]
It is not working for me ( fresh Laravel install v8.5.7, spatie/laravel-ray 1.3.4 and Ray App 1.0.8  on Ubuntu 20.04.1 )

@Tuginho , can you please confirm that you tested it and it is working for you?

Thank you again,
Clement

**Reply by Tuginho** [reply]
On windows with docker + wsl2 it works fine. 

For linux have a look here:
[Support host.docker.internal DNS name to host](https://github.com/docker/for-linux/issues/264)

**Comment by ronfuller** [comment]
After installing on Windows w/WSL 2 and Laravel Sail using the Docker configuration, I only get the following error every 5 seconds:

php_network_getaddresses: getaddrinfo failed in Handler line 47, which is just the exception report

```
parent::report($exception)
```

@Tuginho what docker-compose and ray.php configuration did you use to get it to work on Windows?

**Reply by Tuginho** [reply]
additional: It's running php7.4 and docker engine v20.10.2 for me.

Nothing changed on docker-compose, just added host.docker.internal in ray.php:

`<?php

return [
    /*
    * This settings controls whether data should be sent to Ray.
    *
    * By default, `ray()` will only transmit data in non-production environments.
    */
    'enable' => true,

    /*
    * When enabled, all things logged to the application log
    * will be sent to Ray as well.
    */
    'send_log_calls_to_ray' => true,

    /*
    * When enabled, all things passed to `dump` or `dd`
    * will be sent to Ray as well.
    */
    'send_dumps_to_ray' => true,

    /*
    * The host used to communicate with the Ray app.
     *
    * For usage in Docker on Mac or Windows, you can replace host with 'host.docker.internal'
     *
    * For usage in Homestead on Mac or Windows, you can replace host with '10.0.2.2'
    */
    'host' => 'host.docker.internal',

    /*
    * The port number used to communicate with the Ray app.
    */
    'port' => 23517,

    /*
     * Absolute base path for your sites or projects in Homestead,
     * Vagrant, Docker, or another remote development server.
     */
    'remote_path' => null,

    /*
     * Absolute base path for your sites or projects on your local
     * computer where your IDE or code editor is running on.
     */
    'local_path' => null,
];
`

**Reply by ronfuller** [reply]
I was able to get it working with that setting. Are you able to use the feature to launch the file in your IDE? I am using VSCode and if i set the local_path to "\\\\wsl$\\Ubuntu-20.04\\home\\[username]\\[project-root]  the file path is built with forward slash in Ray which doesn't allow Windows to launch the file correctly in VSCode.

So the ApplicationsController file gets referenced in Ray as:
\\wsl$\Ubuntu-20.04\home\ronfu\portal-api/app/Http/Controllers/ApplicationsController.php

**Comment by lilisgeorge** [comment]
Hi,

Just wanted to let you know that if you are running ray on Ubuntu + Sail then you should try using port `172.17.0.1` instead of 'host.docker.internal' which works perfectly on Windows OS.

Regards,

**Comment by Kiota2019** [comment]
U can use the following command to publish a config file for docker

**php artisan ray:publish-config --docker**
# or
php artisan ray:publish-config --homestead
